pkg/otlp/model/translator/config.go

```diff
@@ -35,7 +39,7 @@ type translatorConfig struct {
 
 	// hostname provider configuration
 	previewHostnameFromAttributes bool
-	fallbackHostnameProvider      HostnameProvider
+	fallbackSourceProvider        source.Provider
```

Comment:
With the name change on the interface, this could be shortened to `fallbackSourcer` without losing brevity.